### PR TITLE
MULE-10134: Migrate extensions to use isolated test runner - Migrate …

### PR DESCRIPTION
**Pull request by gsfernandes**
…HTTP extension to use isolated test runner

### CODE REVIEW

File: tests/http/src/test/java/org/mule/test/module/http/functional/AbstractHttpTestCase.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.test.module.http.functional;
+
+import org.mule.extension.http.internal.request.validator.HttpRequesterConfig;
+import org.mule.extension.http.internal.request.validator.HttpRequesterProvider;
+import org.mule.functional.junit4.MuleArtifactFunctionalTestCase;
+import org.mule.functional.junit4.runners.ArtifactClassLoaderRunnerConfig;
+
+@ArtifactClassLoaderRunnerConfig(exportClasses = {HttpRequesterProvider.class, HttpRequesterConfig.class})
```
Comment (pablokraan) [inline tests/http/src/test/java/org/mule/test/module/http/functional/AbstractHttpTestCase.java]: Why is required to export this classes?

File: tests/http/src/test/java/org/mule/test/module/http/functional/AbstractClientTlsRestrictedTlsProtocolsAndCiphersTestCase.java [hunk 2]
```diff
@@ -35,7 +38,13 @@
 import org.junit.Rule;
 import org.junit.Test;
 
-public abstract class AbstractClientTlsRestrictedTlsProtocolsAndCiphersTestCase extends AbstractHttpTestCase
+/**
+ * Needs to access resources for validating different TLS scenarios and it cannot be done with
```
Comment (pablokraan) [inline tests/http/src/test/java/org/mule/test/module/http/functional/AbstractClientTlsRestrictedTlsProtocolsAndCiphersTestCase.java]: Create a abstract class to avoid duplicating this javadoc and the getAnnotatedExtensionClasses in the two classes that require this changes.

File: tests/http/src/test/java/org/mule/test/module/http/functional/HttpListenerStaticResourcesTestCase.java [hunk 3]
```diff
@@ -113,6 +114,7 @@ public void contentTypeShouldBeDeterminedFromResource() throws Exception
         assertResponseContentType("image/gif");
     }
 
+    @Ignore("It is not supported to access application resources from extension classloader")
```
Comment (pablokraan) [inline tests/http/src/test/java/org/mule/test/module/http/functional/HttpListenerStaticResourcesTestCase.java]: Add the reference to a JIRA so the exclusion can be removed when is fixed

File: tests/http/src/test/resources/log4j2-test.xml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pablokraan) [inline tests/http/src/test/resources/log4j2-test.xml]: Is this file required?

File: tests/http/src/test/java/org/mule/test/module/http/functional/AbstractHttpTestCase.java (hunk 1 above)
Comment (gsfernandes) [inline tests/http/src/test/java/org/mule/test/module/http/functional/AbstractHttpTestCase.java]: There are two tests that use those classes, both validate if the connector is well configured so it asserts those objects.

File: tests/http/src/test/resources/log4j2-test.xml (hunk 4 above)
Comment (gsfernandes) [inline tests/http/src/test/resources/log4j2-test.xml]: No, just added for debugging the class loaders built for the test.
